tiledb/api/src/array/attribute.rs

```diff
@@ -498,6 +478,85 @@ impl<'ctx> Builder<'ctx> {
     }
 }
 
+/// Encapsulation of data needed to construct an Attribute's fill value
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct FillData {
+    pub data: serde_json::value::Value,
+    pub nullability: Option<bool>,
+}
+
+/// Encapsulation of data needed to construct an Attribute
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct AttributeData {
+    pub name: String,
+    pub datatype: Datatype,
+    pub nullability: Option<bool>,
+    pub cell_val_num: Option<u32>,
```

Comment:
Is the idea behind the Option's that we want to test when some of the builder methods are **not** called?

Reply:
It's not a required field for the Thing, so I decided to write that down here.  Similar to the decision for the Builder whether to have a field be a `new` argument or a setter method